packages/core/src/components/dialog/multistepDialog.tsx

```diff
@@ -0,0 +1,236 @@
+        const steps = this.getStepChildren();
+        if (steps.length <= this.state.selectedIndex) {
+            return null;
+        }
+
+        const stepProp = steps[this.state.selectedIndex].props;
+        if (stepProp.renderPanel === undefined) {
+            return null;
+        }
+
+        const { className, panelClassName } = stepProp;
+        return (
+            <div
+                className={classNames(Classes.MULTISTEP_DIALOG_RIGHT_PANEL, className, panelClassName)}
+                role="steppanel"
+            >
+                {stepProp.renderPanel({ updateDialog: this.updateDialog })}
+                {this.renderFooter()}
+            </div>
+        );
+    }
+
+    private updateDialog = (state: { enableNextButton: boolean }) => {
+        this.setState({
+            enableNextButton: state.enableNextButton,
+        });
+    };
+
+    private renderFooter() {
+        return (
+            <div className={Classes.MULTISTEP_DIALOG_FOOTER}>
+                <div className={Classes.DIALOG_FOOTER_ACTIONS}>{this.renderButtons()}</div>
+            </div>
+        );
+    }
+
+    private renderButtons() {
+        const buttons = [];
+        if (this.state.selectedIndex > 0) {
+            buttons.push(<Button key="back" onClick={this.handleClickBack()} text="Back" />);
+        }
+
+        if (this.state.selectedIndex === this.getStepChildren().length - 1) {
+            buttons.push(this.getFinalButton());
+        } else {
+            buttons.push(
+                <Button
+                    disabled={!this.state.enableNextButton}
+                    intent="primary"
+                    key="next"
+                    onClick={this.handleClickNext()}
+                    text="Next"
+                />,
+            );
+        }
+
+        return buttons;
+    }
+
+    private handleClickBack() {
+        return this.handleChangeStep(this.state.selectedIndex - 1);
+    }
+
+    private handleClickNext() {
+        return this.handleChangeStep(this.state.selectedIndex + 1);
+    }
+
+    private handleChangeStep(index: number) {
+        return (event: React.MouseEvent<HTMLElement>) => {
+            if (this.props.onChange !== undefined) {
+                const steps = this.getStepChildren();
+                const prevStepId = steps[this.state.selectedIndex].props.id;
+                const newStepId = steps[index].props.id;
+                this.props.onChange(newStepId, prevStepId, event);
+            }
+            this.setState({
+                lastViewedIndex: Math.max(this.state.lastViewedIndex, index),
+                selectedIndex: index,
+            });
+        };
+    }
+
+    private getFinalButton() {
+        if (this.props.finalButtonProps !== undefined) {
+            return <Button key="final" {...this.props.finalButtonProps} />;
+        } else {
+            return <Button intent="danger" key="final" onClick={this.props.onClose} text="Submit" />;
+        }
```

Comment:
```suggestion
return <Button intent="primary" key="final" onClick={this.props.onClose} text="Submit" {...this.props.finalButtonProps} />
```
Similar to how `buttonProps` are handled in `TimezonePicker`

Also, don't think the submit button should default to intent "danger"